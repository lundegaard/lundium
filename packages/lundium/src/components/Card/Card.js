import React from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';

import classNamesByBreakpoint from '../../utils/classNamesByBreakpoint';
import Box from '../Box';

const boolPropToClassName = prefix =>
	classNamesByBreakpoint((value, breakpoint) => {
		if (value === true) {
			return breakpoint === 'xs' ? `${prefix}` : `${prefix}-${breakpoint}`;
		}

		return breakpoint === 'xs'
			? `${prefix}-${value}`
			: `${prefix}-${breakpoint}-${value}`;
	});

const Card = ({ className, boxShadow, borderRadius, ...rest }) => (
	<Box
		{...rest}
		className={cx(
			boolPropToClassName('shadow')(boxShadow),
			boolPropToClassName('border-radius')(borderRadius),
			className,
		)}
	/>
);

const boxShadowPropType = PropTypes.oneOf([true, 'sm', 'lg', 'none']);
const borderRadiusPropType = PropTypes.oneOf(['none', true]);

Card.propTypes = {
	borderRadius: PropTypes.oneOfType([
		PropTypes.object,
		boxShadowPropType,
		PropTypes.arrayOf(borderRadiusPropType),
	]),
	boxShadow: PropTypes.oneOfType([
		PropTypes.object,
		boxShadowPropType,
		PropTypes.arrayOf(boxShadowPropType),
	]),
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	/** Class name for root. */
	className: PropTypes.string,
};

export default Card;
