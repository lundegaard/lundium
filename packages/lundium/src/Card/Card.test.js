import React from 'react';
import { shallow } from 'enzyme';

import Card from './Card';

describe('<Card />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<Card {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
