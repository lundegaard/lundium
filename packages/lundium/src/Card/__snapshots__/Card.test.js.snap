// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> matches snapshot when shallowly rendered 1`] = `
<forwardRef(Box)
  className=""
/>
`;
